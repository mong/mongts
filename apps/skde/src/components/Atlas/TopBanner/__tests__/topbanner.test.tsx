/**
 * To view a rendered version of all the snapshots:
 *
 * Copy the contents of ./__snapshots__/index.tsx.snap
 * Go to https://codesandbox.io/s/snapshot-viewer-lqjjsm?file=/src/snapshots.js
 * Replace the contents with what you copied
 */

import { render } from "@testing-library/react";
import { TopBanner } from "..";
import { test, expect } from "vitest";

test("BokmÃ¥l render", async () => {
  const title = "Dette er en tittel";
  const { container } = render(
    <TopBanner mainTitle={title} ia={false} lang="nb" />,
  );
  expect(container).toMatchSnapshot();
});

test("Nynorsk render", async () => {
  const title = "Dette er ein tittel";
  const { container } = render(
    <TopBanner mainTitle={title} ia={false} lang="nn" />,
  );
  expect(container).toMatchSnapshot();
});

test("English render", async () => {
  const title = "This is a title";
  const { container } = render(
    <TopBanner mainTitle={title} ia={false} lang="en" />,
  );
  expect(container).toMatchSnapshot();
});

test("Random lang render", async () => {
  const title = "Dette er ein tittel";
  const { container } = render(
    <TopBanner mainTitle={title} ia={false} lang="qwerty" />,
  );
  expect(container).toMatchSnapshot();
});

test("IA render", async () => {
  const title = "Dette er en tittel";
  const { container } = render(
    <TopBanner mainTitle={title} ia={true} lang="nb" />,
  );
  expect(container).toMatchSnapshot();
});

test("IA english render", async () => {
  const title = "Dette er en tittel";
  const { container } = render(
    <TopBanner mainTitle={title} ia={true} lang="en" />,
  );
  expect(container).toMatchSnapshot();
});
