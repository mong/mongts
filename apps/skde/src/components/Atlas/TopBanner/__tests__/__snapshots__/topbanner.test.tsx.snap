// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Bokm√•l render 1`] = `
<div>
  <div
    class="_atlasTopBanner_a3e095"
  >
    <div
      class="_bannerWrapper_a3e095"
    >
      <nav
        aria-label="breadcrumb"
        class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
      >
        <ol
          class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-r3y2q9-MuiTypography-root-MuiLink-root"
              href="/helseatlas/"
            >
              Forside
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              color="#D3D7DA"
              data-testid="NavigateNextIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <div
              class="_breadcrumb_text_a3e095"
            >
              Dette er en tittel
            </div>
          </li>
        </ol>
      </nav>
      <div
        class="_rightbanner_a3e095"
      />
    </div>
  </div>
</div>
`;

exports[`English render 1`] = `
<div>
  <div
    class="_atlasTopBanner_a3e095"
  >
    <div
      class="_bannerWrapper_a3e095"
    >
      <nav
        aria-label="breadcrumb"
        class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
      >
        <ol
          class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-r3y2q9-MuiTypography-root-MuiLink-root"
              href="/helseatlas/en/"
            >
              Front page
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              color="#D3D7DA"
              data-testid="NavigateNextIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <div
              class="_breadcrumb_text_a3e095"
            >
              This is a title
            </div>
          </li>
        </ol>
      </nav>
      <div
        class="_rightbanner_a3e095"
      />
    </div>
  </div>
</div>
`;

exports[`IA english render 1`] = `
<div>
  <div
    class="_atlasTopBanner_a3e095"
  >
    <div
      class="_bannerWrapper_a3e095"
    >
      <nav
        aria-label="breadcrumb"
        class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
      >
        <ol
          class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-r3y2q9-MuiTypography-root-MuiLink-root"
              href="/helseatlas/en/"
            >
              Front page
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              color="#D3D7DA"
              data-testid="NavigateNextIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <div
              class="_breadcrumb_text_a3e095"
            >
              Dette er en tittel
            </div>
          </li>
        </ol>
      </nav>
      <div
        class="_rightbanner_a3e095"
      >
        <div
          class="_map_button_a3e095"
        >
          <a
            href="ia"
          >
            Map
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`IA render 1`] = `
<div>
  <div
    class="_atlasTopBanner_a3e095"
  >
    <div
      class="_bannerWrapper_a3e095"
    >
      <nav
        aria-label="breadcrumb"
        class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
      >
        <ol
          class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-r3y2q9-MuiTypography-root-MuiLink-root"
              href="/helseatlas/"
            >
              Forside
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              color="#D3D7DA"
              data-testid="NavigateNextIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <div
              class="_breadcrumb_text_a3e095"
            >
              Dette er en tittel
            </div>
          </li>
        </ol>
      </nav>
      <div
        class="_rightbanner_a3e095"
      >
        <div
          class="_map_button_a3e095"
        >
          <a
            href="ia"
          >
            Kart
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Nynorsk render 1`] = `
<div>
  <div
    class="_atlasTopBanner_a3e095"
  >
    <div
      class="_bannerWrapper_a3e095"
    >
      <nav
        aria-label="breadcrumb"
        class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
      >
        <ol
          class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-r3y2q9-MuiTypography-root-MuiLink-root"
              href="/helseatlas/"
            >
              Forside
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              color="#D3D7DA"
              data-testid="NavigateNextIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <div
              class="_breadcrumb_text_a3e095"
            >
              Dette er ein tittel
            </div>
          </li>
        </ol>
      </nav>
      <div
        class="_rightbanner_a3e095"
      />
    </div>
  </div>
</div>
`;

exports[`Random lang render 1`] = `
<div>
  <div
    class="_atlasTopBanner_a3e095"
  >
    <div
      class="_bannerWrapper_a3e095"
    >
      <nav
        aria-label="breadcrumb"
        class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
      >
        <ol
          class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-r3y2q9-MuiTypography-root-MuiLink-root"
              href="/helseatlas/"
            >
              Forside
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              color="#D3D7DA"
              data-testid="NavigateNextIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <div
              class="_breadcrumb_text_a3e095"
            >
              Dette er ein tittel
            </div>
          </li>
        </ol>
      </nav>
      <div
        class="_rightbanner_a3e095"
      />
    </div>
  </div>
</div>
`;
